core/trino-main/src/main/java/io/trino/operator/join/PagesHash.java

```diff
@@ -13,224 +13,22 @@
  */
 package io.trino.operator.join;
 
-import io.airlift.units.DataSize;
-import io.trino.operator.HashArraySizeSupplier;
-import io.trino.operator.PagesHashStrategy;
 import io.trino.spi.Page;
 import io.trino.spi.PageBuilder;
-import it.unimi.dsi.fastutil.longs.LongArrayList;
-import org.openjdk.jol.info.ClassLayout;
 
-import java.util.Arrays;
-
-import static io.airlift.slice.SizeOf.sizeOf;
-import static io.airlift.units.DataSize.Unit.KILOBYTE;
-import static io.trino.operator.SyntheticAddress.decodePosition;
-import static io.trino.operator.SyntheticAddress.decodeSliceIndex;
-import static io.trino.util.HashCollisionsEstimator.estimateNumberOfHashCollisions;
-import static java.lang.Math.toIntExact;
-import static java.util.Objects.requireNonNull;
-
-// This implementation assumes arrays used in the hash are always a power of 2
-public final class PagesHash
+public interface PagesHash
```

Comment:
Interface call might prevent methods from being inlined. I wonder what is the thought process around that?

Reply:
Those classes are isolated per query and only one implementation is used so with any luck JIT will easily inline it.